### DISCUSSION TITLE
Openthread sleepy Child re-attach process

### DISCUSSION THREAD

**Discussion by Exormeter** [openthread/openthread#9943] [General]
Hello, I have a question about the timeslot where the sleepy end device is searching for a router in it area.
I am experimenting with the energy consumption and was wondering if the timeslot could be reduced. Openthread certification is currently not a concern for me, so I am free in this regard.

I have found this thread for the subject matter (https://devzone.nordicsemi.com/f/nordic-q-a/48433/openthread-sleepy-child-re-attach-process) and I was able to recompile my project with changes in the values of `kParentResponseMaxDelayRouters`
`kParentResponseMaxDelayAll`, but I could not observe any change in duration for the re-attach interval. The attached screenshot shows the spike in energy consumption for about 3.5 seconds.

My question is, why does the change of the parameters do not change the interval and which parameter could I change to see a difference in behavior?
  
![ppk-20240318T151102](https://github.com/openthread/openthread/assets/5355243/b912c19e-d2d3-4450-b604-3ad35f098a7b)

**Comment by abtink** [comment]
The `kParentResponseMaxDelayAll` and `kParentResponseMaxDelayRouters` values determine how long a child device waits for "Parent Response" messages after sending a "Parent Request".  Parents randomly delay their responses within this window to prevent collisions (multiple responses at once). These are Thread specification-defined constants essential for the attach process. I would highly advice against changing these values

I think you may be interested in the OpenThread-specific "Attach Backoff" feature.

https://github.com/openthread/openthread/blob/8b906eb4679ce811ba32c260ddc0654f73f3b362/src/core/config/mle.h#L137-L220

There are a bunch of config parameters that control the timings of this feature:
- `OPENTHREAD_CONFIG_MLE_ATTACH_BACKOFF_MINIMUM_INTERVAL`
- `OPENTHREAD_CONFIG_MLE_ATTACH_BACKOFF_MAXIMUM_INTERVAL`
- `OPENTHREAD_CONFIG_MLE_ATTACH_BACKOFF_JITTER_INTERVAL`
- `OPENTHREAD_CONFIG_MLE_ATTACH_BACKOFF_DELAY_TO_RESET_BACKOFF_INTERVAL`
